src/helpers/helpers.options.js

```diff
@@ -175,3 +175,64 @@ export function resolve(inputs, context, index, info) {
 		}
 	}
 }
+
+const notIndexable = ['borderDash', 'fill', 'events'];
+const scriptable = (value) => typeof value === 'function';
+const indexable = (key, value) => isArray(value) && notIndexable.indexOf(key) === -1;
+const readKey = (prefix, name) => prefix ? prefix + _capitalize(name) : name;
+const defined = (value) => typeof value !== 'undefined';
+
+/**
+ * Creates a Proxy for resolving raw values for options.
+ * @param {object[]} scopes - The option scopes to look for values, in resolution order
+ * @param {string[]} [prefixes] - The prefixes for values, in resolution order.
+ * @private
+ */
+export function _resolver(scopes, prefixes = ['']) {
+	const cache = Object.create(null);
+	return new Proxy(cache, {
+		get(target, prop) {
+			let value = Reflect.get(target, prop);
```

Comment:
My first thought on this was performance and needing to look up the loop over and over again. `Reflect` is a nice solution to this!